components/net/cookie_storage.rs

```diff
@@ -62,6 +74,12 @@ impl CookieStorage {
 
     // http://tools.ietf.org/html/rfc6265#section-5.3
     pub fn push(&mut self, mut cookie: Cookie, source: CookieSource) {
+        // https://www.ietf.org/id/draft-ietf-httpbis-cookie-alone-01.txt Step 1
+        if cookie.cookie.secure && source == CookieSource::HTTP {
+            // New cookie is not from a "secure" protocol, ignore.
```

Comment:
CookieSource::HTTP only means that it's an HTTP/HTTPS request, rather than document.cookie. Let's add a ServoUrl argument to CookieSource::HTTP so we can check it here.